src/pages/index.js

```diff
@@ -62,36 +104,106 @@ function handleImageClick(cardData) {
 
 //edit profile
 function handleProfileSubmit(data) {
-  userInfo.setUserInfo({name: data.name, job: data.description});
-editPopup.close(); }
-
-//add card
+  console.log(data);
+  api.updateUserInfo({
+    name: data.name,
+    about: data.about
+  })
+    .then(updatedUserData => {
+      user.setUserInfo({
+        name: updatedUserData.name, 
+        about: updatedUserData.about || updatedUserData.dascription || updatedUserData.bio
+      });
+      editPopup.close();
+    })
+    .catch(err => console.error(`Error updating profile: ${err}`));
+   }
+
+// Add card
 function handleAddCardSubmit(data) {
-  const card = {
+  const cardData = {
     name: data.title,
     link: data.url
   };
-  const cardElement = createCard(card);
-  section.addItems(cardElement);
-  addcardPopup.close();
-  profileAddFormElement.reset();
-  addFormValidator.disableButton();
+
+  // Create the Card 
+  api.addNewCard(cardData)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      section.addItems(cardElement);  
+      addCardPopup.close();            
+      profileAddFormElement.reset();  
+      addFormValidator.disableButton();  
+    })
+    .catch(err => {
+      console.error(`Error adding card: ${err}`);
+    });
+}
+
+//avatar
+function handleAvatarSubmit(data){
+  api.updateUserAvatar({
+    avatar: data.url
+  })
+    .then(updateUserData => {
+      user.setUserInfo({
+        avatar: updateUserData.avatar
+      });
+      avatarPopup.close();
+    })
+    .catch(err => console.error(`Error updating avatar: ${err}`));
+}
+
+//delete
+function handleDeleteSubmit(card){
+  if(card){const confirmButton = document.querySelector('#delete-confirm-button');
+  deletePopup.open();
+  confirmButton.addEventListener('click', function onConfirmClick() {
+    card.removeCard();
+    api.deleteCard(card._id);
+    deletePopup.close();
+    confirmButton.removeEventListener('click', onConfirmClick)
+  })}
+  
+}
+
+//like
+function handleCardLike(card){
+  const isLiked = card.isLiked(); 
+
+  const likePromise = isLiked
+    ? api.removeLike(card.getId())  
+    : api.addLike(card.getId()); 
+
+  likePromise
+    .then(response =>  response.json())
+    .then(updatedCardData => {
+      card.updateLikes(updatedCardData.isLiked);  
+    })
+    .catch(err => console.error(`Error updating like status: ${err}`));
 }
 
-function openEditProfileModal() {
-  const currentUser = userInfo.getUserInfo();
+function openProfileModal(popup) {
+  const currentUser = user.getUserInfo();
   userNameInput.value = currentUser.name;
-  userJobInput.value = currentUser.job;
-  editPopup.open();
+  userJobInput.value = currentUser.about;
+  userAvatarInput.value = currentUser.avatar;
+  popup.open();
 }
 
 // Event Listeners
-profileEditButton.addEventListener('click', openEditProfileModal);
+avatar.addEventListener('click', () =>{openProfileModal(avatarPopup)})
```

Comment:
`openProfileModal` is only for the profile modal opening


Please, don't use it here

Reply:
Fixed